### PR TITLE
[CALCITE-2995] Implement the DAYNAME，MONTHNAME function

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
`MONTHNAME(date)`
 Returns the full name of the month for date. The language used for the name is controlled by the system time config.

SQL:

```
SELECT MONTHNAME('2018-01-01') monthname;
```

Result:

| monthname |
| --------- |
| January   |

`DAYNAME(date)`
 Returns the name of the weekday for date. The language used for the name is controlled by the system time config.

SQL:

```
SELECT DAYNAME('2018-01-01') dayname;
```

Result:

| dayname |
| ------- |
| Monday  |

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java [hunk 1]
```diff
@@ -5761,6 +5761,88 @@ private void checkNullOperand(SqlTester tester, String op) {
...
+    Locale.setDefault(Locale.FRENCH);
+    tester.checkScalar("monthname(DATE '2019-02-10')",
+            "février", "VARCHAR(2000) NOT NULL");
+    tester.checkScalar("monthname(TIMESTAMP '2019-02-10 02:10:12')",
+            "février", "VARCHAR(2000) NOT NULL");
+
+    // finally, set the default to Chinese
+    Locale.setDefault(Locale.SIMPLIFIED_CHINESE);
+  }
+
```
Comment (hsyuan) [inline core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java]: Can you add test case for date before epoch?

**Comment by chunweilei** [comment]
I left some comments in the JIRA.

**Comment by XuQianJin-Stars** [comment]
> I left some comments in the JIRA.

well, I left some comments in the JIRA,too.

**Comment by XuQianJin-Stars** [comment]
hi @hsyuan @chunweilei Thank you very much，Comments addressed.

File: site/_docs/reference.md [hunk 2]
```diff
@@ -1510,10 +1510,6 @@ Not implemented:
 | {fn TIMESTAMPADD(timeUnit, count, datetime)} | Adds an interval of *count* *timeUnit*s to a datetime
 | {fn TIMESTAMPDIFF(timeUnit, timestamp1, timestamp2)} | Subtracts *timestamp1* from *timestamp2* and returns the result in *timeUnit*s
 
-Not implemented:
-
-* {fn DAYNAME(date)}
-* {fn MONTHNAME(date)}
```
Comment (ambition119) [inline site/_docs/reference.md]: JDBC String function` {fn DAYNAME(date)}` and `{fn RIGHT(string, integer)}` not create in` SqlJdbcFunctionCall `and `SqlOperatorBaseTest.testJdbcFn()` not test.

Comment (XuQianJin-Stars) [reply site/_docs/reference.md]: hi @ambition119  Thank you very much.

**Comment by XuQianJin-Stars** [comment]
The AppVeyor's test is broken.

**Comment by XuQianJin-Stars** [comment]
The CI is broken.

**Comment by XuQianJin-Stars** [comment]
hi @hsyuan @julianhyde Thank you very much，Comments addressed.

**Comment by julianhyde** [comment]
I've added comments in JIRA.

**Comment by XuQianJin-Stars** [comment]
hi @julianhyde @hsyuan Thank you very much，Comments addressed.I have modified this PR.

**Comment by XuQianJin-Stars** [comment]
The CI is broken.

**Comment by XuQianJin-Stars** [comment]
hi @julianhyde @hsyuan Can you continue to take up your valuable time to review this pr?

**Pull request closed without merging**